src/pmfs/include/pmfs.h

```diff
@@ -0,0 +1,186 @@
+#define PMFS_MAX_RECURSION	40
+
+/** struct holding attributes for a PMFS container */
+struct  pmfs_attr {
+	/** Optional user ID for DFS container. */
+	uint64_t da_id;
+	/** Default Chunk size for all files in container */
+	daos_size_t da_chunk_size;
+	/** Default Object Class for all objects in the container */
+	uint32_t da_oclass_id;
+	/*
+	 * Consistency mode for the PMFS container: PMFS_RELAXED, PMFS_BALANCED.
+	 * If set to 0 or more generally not set to balanced explicitly, relaxed
+	 * mode will be used. In the future, Balanced mode will be the default.
+	 */
+	uint32_t da_mode;
+};
+
+/** object struct that is instantiated for a PMFS open object */
+struct pmfs_obj {
+	/* Reference number */
+	int			ref;
+	/** DAOS object ID */
+	daos_obj_id_t		oid;
+	/** mode_t containing permissions & type */
+	mode_t			mode;
+	/** open access flags */
+	int			flags;
+	/** DAOS object ID of the parent of the object */
+	daos_obj_id_t		parent_oid;
+	/** entry name of the object in the parent */
+	char			name[PMFS_MAX_NAME + 1];
+	/** File size */
+	daos_size_t		file_size;
+	/** Symlink value if object is a symbolic link */
+	char			*value;
+	/** Default chunk size for all entries in dir */
+	daos_size_t		chunk_size;
+};
+
+/** pmfs struct that is instantiated for a mounted PMFS namespace */
+struct pmfs {
+	/** flag to indicate whether the dfs is mounted */
+	bool			mounted;
+	/** flag to indicate whether pmfs is mounted with balanced mode (DTX) */
+	bool			use_dtx;
+	/** lock for threadsafety */
+	pthread_mutex_t		lock;
+	/** uid - inherited from container. */
+	uid_t			uid;
+	/** gid - inherited from container. */
+	gid_t			gid;
+	/** Access mode (RDONLY, RDWR) */
+	int			amode;
+	/** Open pool handle of the DFS */
+	daos_handle_t		poh;
+	/** Open container handle of the DFS */
+	daos_handle_t		coh;
+	/** Object ID reserved for this DFS (see oid_gen below) */
+	daos_obj_id_t		oid;
+	/** superblock object OID */
+	daos_obj_id_t		super_oid;
+	/** Root object info */
+	struct pmfs_obj		root;
+	/** PMFS container attributes (Default chunk size, etc.) */
+	struct pmfs_attr	attr;
+	/** Task ring list */
+	struct spdk_ring	*task_ring;
+};
+
+struct pmfs_entry {
+	/** mode (permissions + entry type) */
+	mode_t			mode;
+	/* Length of value string, not including NULL byte */
+	uint16_t		value_len;
+	/** Object ID if not a symbolic link */
+	daos_obj_id_t		oid;
+	/* Time of last access */
+	time_t			atime;
+	/* Time of last modification */
+	time_t			mtime;
+	/* Time of last status change */
+	time_t			ctime;
+	/** chunk size of file or default for all files in a dir */
+	daos_size_t		chunk_size;
+	/** size of regular file */
+	daos_size_t		file_size;
+	/** Sym Link value */
+	char			*value;
+};
+
+int pmfs_mkfs(daos_handle_t poh, uuid_t uuid);
+int pmfs_mount(daos_handle_t poh, daos_handle_t coh, int flags, struct pmfs **pmfs);
+int pmfs_umount(struct pmfs *pmfs);
+int pmfs_mkdir(struct pmfs *pmfs, struct pmfs_obj *parent, const char *name,
```

Comment:
Why are these repeated here from src/include/pmfs/pmfs.h ?

Reply:
I  followed other header files  and found they are the same copies , maybe  need using  code to copy.